### ISSUE TITLE
AddImport code fix provider encountered an error

### ISSUE THREAD

**Issue by rchande** [dotnet/roslyn#18256]
```C#
System.ArgumentOutOfRangeException : Index was out of range. Must be non-negative and less than the size of the collection.
Parameter name: index
   at System.ThrowHelper.ThrowArgumentOutOfRangeException(ExceptionArgument argument,ExceptionResource resource)
   at Microsoft.CodeAnalysis.SymbolKey.TypeParameterOrdinalSymbolKey.Resolve(SymbolKeyReader reader)
   at Microsoft.CodeAnalysis.SymbolKey.SymbolKeyReader.ReadWorker(SymbolKeyType type)
   at Microsoft.CodeAnalysis.SymbolKey.SymbolKeyReader.ReadSymbolKeyWorker(Boolean first)
   at Microsoft.CodeAnalysis.SymbolKey.SymbolKeyReader.ReadSymbolKey()
   at Microsoft.CodeAnalysis.SymbolKey.Reader`1.ReadArray[T](Func`1 readFunction)
   at Microsoft.CodeAnalysis.SymbolKey.NamedTypeSymbolKey.Resolve(SymbolKeyReader reader)
   at Microsoft.CodeAnalysis.SymbolKey.SymbolKeyReader.ReadWorker(SymbolKeyType type)
   at Microsoft.CodeAnalysis.SymbolKey.SymbolKeyReader.ReadSymbolKeyWorker(Boolean first)
   at Microsoft.CodeAnalysis.SymbolKey.SymbolKeyReader.ReadSymbolKey()
   at Microsoft.CodeAnalysis.SymbolKey.Reader`1.ReadArray[T](Func`1 readFunction)
   at Microsoft.CodeAnalysis.SymbolKey.MethodSymbolKey.Resolve(SymbolKeyReader reader)
   at Microsoft.CodeAnalysis.SymbolKey.SymbolKeyReader.ReadWorker(SymbolKeyType type)
   at Microsoft.CodeAnalysis.SymbolKey.SymbolKeyReader.ReadSymbolKeyWorker(Boolean first)
   at Microsoft.CodeAnalysis.SymbolKey.Resolve(String symbolKey,Compilation compilation,Boolean ignoreAssemblyKey,CancellationToken cancellationToken)
   at Microsoft.CodeAnalysis.FindSymbols.SymbolFinder.<>c__DisplayClass21_0.<GetUnfilteredSymbolsAsync>b__0(ISymbol s)
   at System.Linq.Enumerable.WhereSelectEnumerableIterator`2.MoveNext()
   at System.Linq.Enumerable.WhereEnumerableIterator`1.MoveNext()
   at System.Linq.Buffer`1..ctor(IEnumerable`1 source)
   at System.Linq.Enumerable.ToArray[TSource](IEnumerable`1 source)
   at System.Collections.Immutable.ImmutableArray.CreateRange[T](IEnumerable`1 items)
   at System.Collections.Immutable.ImmutableArray.ToImmutableArray[TSource](IEnumerable`1 items)
   at async Microsoft.CodeAnalysis.FindSymbols.SymbolFinder.GetUnfilteredSymbolsAsync(<Unknown Parameters>)
   at async Microsoft.CodeAnalysis.FindSymbols.SymbolFinder.AddDeclarationsAsync(<Unknown Parameters>)
   at async Microsoft.CodeAnalysis.FindSymbols.SymbolFinder.FindDeclarationsAsyncImpl(<Unknown Parameters>)
   at async Microsoft.CodeAnalysis.CodeFixes.AddImport.AbstractAddImportCodeFixProvider`1.SearchScope.FindDeclarationsAsync[TSimpleNameSyntax](<Unknown Parameters>)
   at async Microsoft.CodeAnalysis.CodeFixes.AddImport.AbstractAddImportCodeFixProvider`1.SymbolReferenceFinder.GetNamespacesForMatchingExtensionMethodsAsync[TSimpleNameSyntax](<Unknown Parameters>)
   at async Microsoft.CodeAnalysis.CodeFixes.AddImport.AbstractAddImportCodeFixProvider`1.SymbolReferenceFinder.DoAsync[TSimpleNameSyntax](<Unknown Parameters>)
   at async Microsoft.CodeAnalysis.CodeFixes.AddImport.AbstractAddImportCodeFixProvider`1.FindResultsInAllSymbolsInStartingProjectAsync[TSimpleNameSyntax](<Unknown Parameters>)
   at async Microsoft.CodeAnalysis.CodeFixes.AddImport.AbstractAddImportCodeFixProvider`1.FindResultsAsync[TSimpleNameSyntax](<Unknown Parameters>)
   at async Microsoft.CodeAnalysis.CodeFixes.AddImport.AbstractAddImportCodeFixProvider`1.FindResultsAsync[TSimpleNameSyntax](<Unknown Parameters>)
   at async Microsoft.CodeAnalysis.CodeFixes.AddImport.AbstractAddImportCodeFixProvider`1.RegisterCodeFixesAsync[TSimpleNameSyntax](<Unknown Parameters>)
   at async Microsoft.CodeAnalysis.CodeFixes.CodeFixService.GetCodeFixesAsync(<Unknown Parameters>)
   at async Microsoft.CodeAnalysis.Extensions.IExtensionManagerExtensions.PerformFunctionAsync[T](<Unknown Parameters>)
   at Microsoft.VisualStudio.Telemetry.WindowsErrorReporting.WatsonReport.GetClrWatsonExceptionInfo(Exception exceptionObject)
```

Dump at `\\mlangfs1\public\rchande\addimport.dmp`

**Comment by rchande** [comment]
cc @CyrusNajmabadi

**Comment by CyrusNajmabadi** [comment]
1. do you know what youwere typing?  
2. is this a dump of a roslyn you were f5ing? or of a real roslyn build?

**Comment by rchande** [comment]
1. real roslyn build
2. Was trying to add a using to `System.Linq`, typing `.Any()` after an array reference.

**Comment by rchande** [comment]
This was in 26323.1-Preview d15rel Internal Preview

**Comment by CyrusNajmabadi** [comment]
This is a dupe of https://github.com/dotnet/roslyn/issues/17056

@Pilchie Would we want to take the fix (very simple and easy) for 15.2

**Comment by CyrusNajmabadi** [comment]
Looks like it's not even in the top 50 NFWs we have.  closing out.
